src/stream_p.cpp

```diff
@@ -120,25 +123,39 @@ void Aulib::Stream_priv::fSdlCallbackImpl(void* /*unused*/, Uint8 out[], int out
             continue;
         }
 
+        int diffMilliseconds = now - stream->d->fPlaybackStartTick;
+        if (diffMilliseconds <= 0) {
+            continue;
+        }
+
         bool has_finished = false;
         bool has_looped = false;
-        int len = 0;
+        int offset = 0;
+        int end = 0;
+
+        if (diffMilliseconds < bufferMilliseconds) {
+            int offsetMilliseconds = bufferMilliseconds - diffMilliseconds;
+            offset = wantedSamples * offsetMilliseconds / bufferMilliseconds;
+            offset -= offset % 2;
```

Comment:
Is this supposed to round down to fit the channel count? If so, this should be:

```
offset -= offset % fAudioSpec.channels;
```

Reply:
Actually, I had encountered errors on some of the sound effects in DevilutionX in the following section of code.
https://github.com/realnc/SDL_audiolib/blob/7979aa98e3845095c1386b4f1c01e59de3d7d3d3/src/stream_p.cpp#L192-L195

I found that even if SDL was only using one channel, it was still possible for left/right volume to be applied to every other channel.  After that, I got a bit paranoid and started thinking that maybe the number of output channels had nothing to do with the size of the mix buffer. So I simply aligned offset to an even index and waited to see what you'd say about it.

Actually, it looks like the left/right volume is based on position, specifically for stereo sound. Does it make sense to apply left/right volume when channel count is 1? If not, then I think you're right that we should be fitting the channel count, and we should be checking the number of channels before applying left/right volume.